src/Diagnostics.DataProviders/ChangeAnalysisClient.cs

```diff
@@ -57,62 +61,106 @@ private HttpClient httpClient
         /// <summary>
         /// Initializes a new instance of the <see cref="ChangeAnalysisClient"/> class.
         /// </summary>
-        public ChangeAnalysisClient(ChangeAnalysisDataProviderConfiguration configuration, string clientObjectId, string clientPrincipalName = "")
+        public ChangeAnalysisClient(ChangeAnalysisDataProviderConfiguration configuration, string requestTrackingId, string clientObjectId, string clientPrincipalName = "")
         {
             clientObjectIdHeader = clientObjectId;
             clientPrincipalNameHeader = clientPrincipalName;
             changeAnalysisEndPoint = configuration.Endpoint;
             apiVersion = configuration.Apiversion;
+            requestId = requestTrackingId;
         }
 
         /// <inheritdoc/>
         public async Task<List<ResourceChangesResponseModel>> GetChangesAsync(ChangeRequest changeRequest)
         {
-            string requestUri = changeAnalysisEndPoint + $"changes?api-version={apiVersion}";
-            object postBody = new
-            {
-                changeRequest.ResourceId,
-                changeRequest.ChangeSetId
-            };
-            string jsonString = await PrepareAndSendRequest(requestUri, postBody, HttpMethod.Post);
-            List<ResourceChangesResponseModel> resourceChangesResponse = JsonConvert.DeserializeObject<List<ResourceChangesResponseModel>>(jsonString);
-            return resourceChangesResponse;
+            try
+            {
+                string requestUri = changeAnalysisEndPoint + $"changes?api-version={apiVersion}";
+                object postBody = new
+                {
+                    changeRequest.ResourceId,
+                    changeRequest.ChangeSetId
+                };
+                string jsonString = await PrepareAndSendRequest(requestUri, postBody, HttpMethod.Post);
+                List<ResourceChangesResponseModel> resourceChangesResponse = JsonConvert.DeserializeObject<List<ResourceChangesResponseModel>>(jsonString);
+                return resourceChangesResponse;
+            }
+            catch (HttpRequestException httpException)
+            {
+                // Its possible that users dont have access to view the change details, log the exception and send empty list of change details.
+                string message = $"HttpRequestException in GetChangesAsync, Message: {httpException.Message} ";
```

Comment:
will a 500 also result in HttpRequestException?
Maybe we check for specific status code for "users not having access" before consuming the whole exception

Reply:
That's what I attempted to do first. I initially handled it with 403 Forbidden Status code, but I wasn't sure if its status code will remain same or not. I can change it to check specifically for 403 - But if there is some other failure such as 500 on this endpoint, do we still want the users to see the initial view?

Reply:
I haven't looked at other PRs you sent Rekha so you may have already answered this in a different PR but quick question, if you are enforcing RBAC, how are you getting the customers auth token and sending it via this HTTP call to possibly get back a 403?

Reply:
@hforeste 403 is received when Applens makes a call to view change details of code scan changes. Since its same detector code running on Portal vs Backend, its possible that we get 403 when viewing change details